src/Silverback.Integration/Messaging/Inbound/ErrorHandling/ErrorPoliciesHelper.cs

```diff
@@ -10,13 +10,32 @@ namespace Silverback.Messaging.Inbound.ErrorHandling
 {
     internal static class ErrorPoliciesHelper
     {
-        public static async Task<bool> ApplyErrorPoliciesAsync(ConsumerPipelineContext context, Exception exception)
+        public static async Task<bool> ApplyErrorPoliciesAsync(
+            ConsumerPipelineContext context,
+            Exception exception)
         {
-            var failedAttempts = context.Consumer.IncrementFailedAttempts(context.Envelope);
+            int failedAttempts = context.Consumer.IncrementFailedAttempts(context.Envelope);
 
             context.Envelope.Headers.AddOrReplace(DefaultMessageHeaders.FailedAttempts, failedAttempts);
+            context.Envelope.Headers.AddOrReplace(DefaultMessageHeaders.InvalidReason, exception.Message);
```

Comment:
I'm unsure whether `exception.Message` is the right thing to write.
It could disclose some information like internal names etc., which should probably not be an issue since these topics will be private.
But beside that I wonder if a simple `NullReferenceException in YourSubscriber` (`$"{exception.GetType().FullName} in {exception.Source}"`) wouldn't be a better choice?

Reply:
I adjusted it based on your proposal. 👍